Adding and then removing an Empty value does not update the analyses

boutinb opened the pull request:
If you add an empty value, and an analysis that uses a column that is
changed by this new empty value, is updated.
When you remove afterwards this empty value, the analysis is not
changed anymore

JorisGoosen commented on JASP-Desktop/data/datasetpackage.h:
```diff
@@ -107,7 +107,7 @@ class DataSetPackage : public QAbstractItemModel //Not QAbstractTableModel becau
 		const	std::string		&	warningMessage()					const	{ return _warningMessage;						  }
 		const	Version			&	archiveVersion()					const	{ return _archiveVersion;						   }
-		const	emptyValsType	&	emptyValuesMap()					const	{ return _emptyValuesMap;   						}
+				emptyValsType	&	emptyValuesMap()							{ return _emptyValuesMap;   						}
```
@boutinb this kind of breaks the whole "datasetpackage does everything"-paradigm, doesn't it?

Because this way the emptyValuesMap could be changed at any time anywhere right? And nothing would be triggered, not packageModified, not modelChanged, nothing...

boutinb replied:
OK, I have set back the const, and change the emptyValuesMap in DataSetPackage, not in DataSet anymore.

JorisGoosen reviewed (changes requested):
Can you not change it by finding the right place to update it instead of passing non-const references around?

boutinb committed:
Adding and then removing an Empty value does not update the analyses

If you add an empty value, and an analysis that uses a column that is
changed by this new empty value, is updated.
When you remove afterwards this empty value, the analysis is not
changed anymore

JorisGoosen commented:
Code looks nice now, I'll just check if it actually works now and after that Ill merge it.
